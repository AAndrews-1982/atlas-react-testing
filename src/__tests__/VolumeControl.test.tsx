// Imports for testing
import { render } from "@testing-library/react";
import { expect, test } from "vitest";
import { VolumeControl } from "../components/VolumeControl";
import { ContextProvider } from "../components/AppContext";

test("VolumeControl renders as expected", () => {
    // Render the VolumeControl component within the ContextProvider
    const { asFragment } = render(
        <ContextProvider>
            <VolumeControl />
        </ContextProvider>
    );

    // Compare the rendered output to the snapshot
    expect(asFragment()).toMatchSnapshot();
});
