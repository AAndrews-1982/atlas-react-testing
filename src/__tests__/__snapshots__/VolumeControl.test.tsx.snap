// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VolumeControl renders as expected 1`] = `
<DocumentFragment>
  <div
    class="volume-control flex items-center"
  >
    <button
      aria-label="Toggle Mute"
      class="icon-button flex h-6 w-6 cursor-pointer items-center justify-center rounded-full hover:bg-gray-200 active:bg-gray-300 dark:hover:bg-gray-600 dark:active:bg-gray-700"
    >
      <svg
        class="size-5"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.5 3.75a.75.75 0 0 0-1.264-.546L5.203 7H2.667a.75.75 0 0 0-.7.48A6.985 6.985 0 0 0 1.5 10c0 .887.165 1.737.468 2.52.111.29.39.48.7.48h2.535l4.033 3.796a.75.75 0 0 0 1.264-.546V3.75ZM16.45 5.05a.75.75 0 0 0-1.06 1.061 5.5 5.5 0 0 1 0 7.778.75.75 0 0 0 1.06 1.06 7 7 0 0 0 0-9.899Z"
        />
        <path
          d="M14.329 7.172a.75.75 0 0 0-1.061 1.06 2.5 2.5 0 0 1 0 3.536.75.75 0 0 0 1.06 1.06 4 4 0 0 0 0-5.656Z"
        />
      </svg>
    </button>
    <div
      class="slider-container ml-3 flex-grow"
    >
      <label
        class="sr-only"
        for="volume-slider"
      >
        Volume Control
      </label>
      <input
        class="slider w-full cursor-pointer accent-blue-500 dark:accent-blue-300"
        id="volume-slider"
        max="100"
        min="0"
        type="range"
        value="50"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`VolumeControl renders correctly and interacts as expected 1`] = `
<DocumentFragment>
  <div
    class="volume-control flex items-center"
  >
    <button
      aria-label="Toggle Mute"
      class="icon-button flex h-6 w-6 cursor-pointer items-center justify-center rounded-full hover:bg-gray-200 active:bg-gray-300 dark:hover:bg-gray-600 dark:active:bg-gray-700"
    >
      <svg
        class="size-5"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.5 3.75a.75.75 0 0 0-1.264-.546L5.203 7H2.667a.75.75 0 0 0-.7.48A6.985 6.985 0 0 0 1.5 10c0 .887.165 1.737.468 2.52.111.29.39.48.7.48h2.535l4.033 3.796a.75.75 0 0 0 1.264-.546V3.75ZM16.45 5.05a.75.75 0 0 0-1.06 1.061 5.5 5.5 0 0 1 0 7.778.75.75 0 0 0 1.06 1.06 7 7 0 0 0 0-9.899Z"
        />
        <path
          d="M14.329 7.172a.75.75 0 0 0-1.061 1.06 2.5 2.5 0 0 1 0 3.536.75.75 0 0 0 1.06 1.06 4 4 0 0 0 0-5.656Z"
        />
      </svg>
    </button>
    <div
      class="slider-container ml-3 flex-grow"
    >
      <label
        class="sr-only"
        for="volume-slider"
      >
        Volume Control
      </label>
      <input
        class="slider w-full cursor-pointer accent-blue-500 dark:accent-blue-300"
        id="volume-slider"
        max="100"
        min="0"
        type="range"
        value="50"
      />
    </div>
  </div>
</DocumentFragment>
`;
